XenModel/XenAPI-Extensions/VM.cs

```diff
@@ -426,6 +426,31 @@ public bool CanHaveGpu()
             }
         }
 
+        public bool HasSriovRecommendation()
+        {
+            bool result = false;
+
+            XmlDocument xd = GetRecommendations();
+
+            if (xd == null)
+                return result;
+
+            try
+            {
+                XmlNode xn = xd.SelectSingleNode(@"restrictions/restriction[@field='allow_network_sriov']");
+                if (xn == null || xn.Attributes == null)
+                    return result;
+
+                result = bool.Parse(xn.Attributes["value"].Value);
+            }
+            catch (Exception ex)
+            {
+                log.Error("Error parsing allow-network-sriov on the template.", ex);
+            }
+
+            return result;
+        }
+

```

Comment:
duplicates #1980 ?
Can we merge this two commits in one PR?

Reply:
Yes, thanks.